"""<br>
Homework 06<br>
Topic : Avocado<br>
Author: Weerinphas Chimnam<br>
Student ID: 640631127<br>
Date September 5,2021<br>

"""

# Import Salaries Dataset

In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv("avocado.csv")
df.head(50)

Unnamed: 0        Date  AveragePrice  Total Volume     4046       4225  \
0            0  2015-12-27          1.33      64236.62  1036.74   54454.85   
1            1  2015-12-20          1.35      54876.98   674.28   44638.81   
2            2  2015-12-13          0.93     118220.22   794.70  109149.67   
3            3  2015-12-06          1.08      78992.15  1132.00   71976.41   
4            4  2015-11-29          1.28      51039.60   941.48   43838.39   
5            5  2015-11-22          1.26      55979.78  1184.27   48067.99   
6            6  2015-11-15          0.99      83453.76  1368.92   73672.72   
7            7  2015-11-08          0.98     109428.33   703.75  101815.36   
8            8  2015-11-01          1.02      99811.42  1022.15   87315.57   
9            9  2015-10-25          1.07      74338.76   842.40   64757.44   
10          10  2015-10-18          1.12      84843.44   924.86   75595.85   
11          11  2015-10-11          1.28      64489.17  1582.03   52677.92   
12          12  2015-10-04          1.31      61007.10  2268.32   49880.67   
13          13  2015-09-27          0.99     106803.39  1204.88   99409.21   
14          14  2015-09-20          1.33      69759.01  1028.03   59313.12   
15          15  2015-09-13          1.28      76111.27   985.73   65696.86   
16          16  2015-09-06          1.11      99172.96   879.45   90062.62   
17          17  2015-08-30          1.07     105693.84   689.01   94362.67   
18          18  2015-08-23          1.34      79992.09   733.16   67933.79   
19          19  2015-08-16          1.33      80043.78   539.65   68666.01   
20          20  2015-08-09          1.12     111140.93   584.63  100961.46   
21          21  2015-08-02          1.45      75133.10   509.94   62035.06   
22          22  2015-07-26          1.11     106757.10   648.75   91949.05   
23          23  2015-07-19          1.26      96617.00  1042.10   82049.40   
24          24  2015-07-12          1.05     124055.31   672.25   94693.52   
25          25  2015-07-05          1.35     109252.12   869.45   72600.55   
26          26  2015-06-28          1.37      89534.81   664.23   57545.79   
27          27  2015-06-21          1.27     104849.39   804.01   76688.55   
28          28  2015-06-14          1.32      89631.30   850.58   55400.94   
29          29  2015-06-07          1.07     122743.06   656.71   99220.82   
30          30  2015-05-31          1.23      95123.62   922.37   70469.69   
31          31  2015-05-24          1.19     101470.91   680.27   71376.81   
32          32  2015-05-17          1.43     109857.47  1150.55   81955.16   
33          33  2015-05-10          1.26     120427.91  1420.43  102000.52   
34          34  2015-05-03          1.20      59197.67   919.87   45490.05   
35          35  2015-04-26          1.22      49585.46   875.65   35841.75   
36          36  2015-04-19          1.19      49064.73   774.15   33941.51   
37          37  2015-04-12          1.13      48364.29   864.27   30374.15   
38          38  2015-04-05          1.16      47362.13   961.77   35577.66   
39          39  2015-03-29          1.02      67799.08  1402.28   58623.22   
40          40  2015-03-22          1.12      46346.85  2141.83   34313.56   
41          41  2015-03-15          1.11      43045.79  2128.26   30447.17   
42          42  2015-03-08          1.07      40507.36   795.68   30370.64   
43          43  2015-03-01          0.99      55595.74   629.46   45633.34   
44          44  2015-02-22          1.07      45675.05  1088.38   35056.13   
45          45  2015-02-15          1.06      41567.62   986.66   30045.51   
46          46  2015-02-08          0.99      51253.97  1357.37   39111.81   
47          47  2015-02-01          0.99      70873.60  1353.90   60017.20   
48          48  2015-01-25          1.06      45147.50   941.38   33196.16   
49          49  2015-01-18          1.17      44511.28   914.14   31540.32   

       4770  Total Bags  

# Using Python Code to find:
## 1) Which region sold the largest amount of avocado ?

In [3]:
def biggest(region,column):
    df_group = df.groupby([region])[[column]].sum()
    df_sorted = df_group.sort_values( by = column, ascending = False)
    df_sorted[column] = df_sorted[column].astype('int64') 
    df_sorted.head()
    return df_sorted.head()

In [4]:
biggest('region','Total Volume')

Total Volume
region                    
TotalUS         5864740181
West            1086779155
California      1028981653
SouthCentral    1011279618
Northeast        713280909

Answer TotalUS sold the largest amount of avocado.

##      In this region, where the biggest lot of sold avocado came from (4046, 4225, 4770) ?

In [5]:
biggest('region','4046')

4046
region                  
TotalUS       2054936231
SouthCentral   535041386
California     398967012
West           398590531
Southeast      339687092

Answer TotalUS has the biggest lot of sold avocado came from 4046 with 2054936231 units.

In [6]:
biggest('region','4225')

4225
region                
TotalUS     2015011554
Northeast    474484663
California   351491401
West         300758923
GreatLakes   267257721

Answer TotalUS has the biggest lot of sold avocado came from 4225 with 2015011554 units.

In [7]:
biggest('region','4770')

4770
region                 
TotalUS       156175183
GreatLakes     50075971
California     31841045
SouthCentral   22395793
West           20356070

Answer TotalUS has the biggest lot of sold avocado came from 4770 with 156175183 units.

# 2)Which region sold the smallest amount of avocado ?

In [8]:
def smallest(region,column):
    df_group = df.groupby([region])[[column]].sum()
    df_sorted = df_group.sort_values( by = column, ascending = True)
    df_sorted[column] = df_sorted[column].astype('int64') 
    df_sorted.head()
    return df_sorted.head()

In [9]:
smallest('region','Total Volume')

Total Volume
region                  
Syracuse        10942667
Boise           14413187
Spokane         15565275
Albany          16067799
Louisville      16097002

Answer Syracuse sold the smallest amount of avocado.

# In this region, where the smallest lot of sold avocado came from (4046, 4225, 4770) ?

In [10]:
smallest('region','4046')

4046
region                  
Syracuse          331788
GrandRapids       493266
BuffaloRochester  600514
Albany            616539
Louisville        709551

Answer Syracuse has the smallest lot of sold avocado came from 4046 with 331788 units.

In [11]:
smallest('region','4225')

4225
region                   
Boise             1170048
Jacksonville      3526740
StLouis           3799460
Nashville         3810509
NewOrleansMobile  4413219

Answer Boise has the smallest lot of sold avocado came from 4225 with 1170048 units.

In [12]:
smallest('region','4770')

4770
region         
StLouis   26488
Roanoke   26532
Syracuse  28110
Tampa     38989
Orlando   48562

Answer StLouis has the smallest lot of sold avocado came from 4770 with 26488 units.

# 3) Which region sold the highest price of avocado in average ?

In [13]:
df_group = df.groupby(['region'])[['AveragePrice']].mean()
df_sorted = df_group.sort_values( by = 'AveragePrice', ascending = False)
df_sorted.head()

AveragePrice
region                           
HartfordSpringfield      1.818639
SanFrancisco             1.804201
NewYork                  1.727574
Philadelphia             1.632130
Sacramento               1.621568

Answer HartfordSpringfield sold the highest price of avocado in average with 1.818639 per unit

# 4) Find the total amount of income (Avg_Price*Total_Volume) of each region.

In [14]:
df['Income'] = df['AveragePrice'] * df['Total Volume']
df_group = df.groupby(['region'])[['Income']].sum()
df_sorted = df_group.sort_values( by = 'Income', ascending = False)
df_sorted['Income'] = df_sorted['Income'].astype('int64') 
df_sorted

Income
region                         
TotalUS              6387593101
California           1121414026
West                 1066833656
Northeast             960007873
SouthCentral          874059289
Southeast             703630581
GreatLakes            688661844
Midsouth              615723838
LosAngeles            484227610
Plains                360036649
NewYork               335195542
SanFrancisco          185834084
BaltimoreWashington   179908448
Chicago               179110584
DallasFtWorth         175609297
Houston               165571340
Denver                145982845
PhoenixTucson         138451462
Boston                126542904
Seattle               125179125
WestTexNewMexico      121565419
MiamiFtLauderdale     118581508
Portland              112864574
Philadelphia           99815825
Sacramento             94808703
Atlanta                93793368
SanDiego               93527102
NorthernNewEngland     89625254
Tampa                  76959063
HartfordSpringfield    71186448
Orlando                69611214
SouthCarolina          69528083
Detroit                69126241
RaleighGreensboro      59837796
LasVegas               54809053
HarrisburgScranton     52954718
NewOrleansMobile       48676263
RichmondNorfolk        47344816
Charlotte              45743042
CincinnatiDayton       44989579
GrandRapids            37851238
StLouis                37439462
Nashville              35726637
Indianapolis           34024736
Jacksonville           33483957
Columbus               31562947
BuffaloRochester       31545094
Roanoke                27409281
Pittsburgh             23194807
Albany                 21766723
Louisville             17495546
Spokane                17156485
Boise                  15346666
Syracuse               15205192

So, the total amount of income (Avg_Price * Total_Volume) of each region as above.

# 5) Let AVOCADO  Average Weight : <br>4046 => 4 ounces, <br>4225 => 9 ounces, <br>4770 => 12 ounces<br><br>5.1)Find the number of sold avocadoes by region ?

In [15]:
df['AVOCADO Weight'] = df['4046'] * 4 + df['4225'] * 9 + df['4770'] * 12
df_group = df.groupby(['region'])[['AVOCADO Weight']].sum()
df_sorted = df_group.sort_values( by = 'AVOCADO Weight', ascending = False)
df_sorted['AVOCADO Weight'] = df_sorted['AVOCADO Weight'].astype('int64') 
df_sorted

AVOCADO Weight
region                             
TotalUS                 28228951116
California               5141383216
West                     4545465285
Northeast                4492126538
SouthCentral             4392965126
GreatLakes               3381099192
Midsouth                 2671463641
Southeast                2214451062
LosAngeles               1978147429
NewYork                  1521703518
Plains                   1389997173
Chicago                  1057207953
SanFrancisco              928716898
DallasFtWorth             917431209
Houston                   894077141
BaltimoreWashington       847052215
PhoenixTucson             839925269
Boston                    678617728
Denver                    595484016
WestTexNewMexico          571549402
NorthernNewEngland        509080825
Sacramento                483035781
Portland                  441953827
SanDiego                  436954285
Seattle                   422417790
Philadelphia              410395890
MiamiFtLauderdale         372026743
HartfordSpringfield       347950224
Atlanta                   293779694
Detroit                   284369868
SouthCarolina             246028716
RaleighGreensboro         245711736
Tampa                     239948844
LasVegas                  231496629
HarrisburgScranton        217161848
Orlando                   209382740
RichmondNorfolk           208627589
CincinnatiDayton          206933470
GrandRapids               196840841
Charlotte                 188639930
Indianapolis              156570498
NewOrleansMobile          149436782
Columbus                  121886535
Albany                    117570322
Nashville                 114796465
Roanoke                   101874020
BuffaloRochester           99309233
Jacksonville               94508614
StLouis                    88556417
Pittsburgh                 85046665
Louisville                 82626175
Spokane                    64118399
Syracuse                   59177550
Boise                      50522423

So, the number of sold avocadoes by region as above.

# 5.2) Which region sold the largest number of avocados ?

In [16]:
df_sorted.head()

AVOCADO Weight
region                      
TotalUS          28228951116
California        5141383216
West              4545465285
Northeast         4492126538
SouthCentral      4392965126

Answer TotalUS sold the largest number of avocados.

# 6) Normally, the customers buy the avocados by unit or in a bags ?

In [17]:
df['Total units'] = df['4046'] + df['4225'] + df['4770']
df = df.groupby(['region'])[['Total units','Total Bags']].sum()
df = df[['Total units','Total Bags']].astype('int64') 
df

Total units  Total Bags
region                                      
Albany                  13387545     2680254
Atlanta                 60044606    28560513
BaltimoreWashington     99407773    35306146
Boise                    9013776     5399412
Boston                  75778508    21495476
BuffaloRochester        11351716    11610751
California             782299460   246682193
Charlotte               23857602    11697942
Chicago                116836909    16865400
CincinnatiDayton        23623368    20897918
Columbus                19976882    10016387
DallasFtWorth          161949535    46469752
Denver                  79945530    58957007
Detroit                 39160134    24260065
GrandRapids             21042629     9169093
GreatLakes             411050704   178575648
HarrisburgScranton      28279603    13529254
HartfordSpringfield     39337525    11333012
Houston                152952802    50215064
Indianapolis            18280507    11980810
Jacksonville            19071876     9718127
LasVegas                36323912    18052993
LosAngeles             339632018   168264528
Louisville               9461715     6635288
MiamiFtLauderdale       74156782    23516442
Midsouth               351835659   156513626
Nashville               22644922    12967166
NewOrleansMobile        31426196    14268940
NewYork                172739971    67994156
Northeast              516292515   196988393
NorthernNewEngland      57627868    13905027
Orlando                 41799064    16861640
Philadelphia            47963566    23875232
PhoenixTucson          156735075    38908236
Pittsburgh              11560883     7245462
Plains                 233341172    77847204
Portland                61047851    49504359
RaleighGreensboro       32526051    15676677
RichmondNorfolk         29722621    12508228
Roanoke                 15451017     9590987
Sacramento              66181479     8982268
SanDiego                66449250    23342668
SanFrancisco           120939266    14890924
Seattle                 58522780    50691398
SouthCarolina           42600600    18153173
SouthCentral           777887186   233392434
Southeast              433710759   181527648
Spokane                  9310899     6254375
StLouis                 17336801    14736030
Syracuse                 6750240     4192427
Tampa                   46507848    19496691
TotalUS               4226122969  1638600779
West                   719705525   367073631
WestTexNewMexico       109225323    35296515

In [18]:
df['Normally buy'] = np.where(df['Total units'] > df['Total Bags'], 'by units', 'by bags')
df

Total units  Total Bags Normally buy
region                                                   
Albany                  13387545     2680254     by units
Atlanta                 60044606    28560513     by units
BaltimoreWashington     99407773    35306146     by units
Boise                    9013776     5399412     by units
Boston                  75778508    21495476     by units
BuffaloRochester        11351716    11610751      by bags
California             782299460   246682193     by units
Charlotte               23857602    11697942     by units
Chicago                116836909    16865400     by units
CincinnatiDayton        23623368    20897918     by units
Columbus                19976882    10016387     by units
DallasFtWorth          161949535    46469752     by units
Denver                  79945530    58957007     by units
Detroit                 39160134    24260065     by units
GrandRapids             21042629     9169093     by units
GreatLakes             411050704   178575648     by units
HarrisburgScranton      28279603    13529254     by units
HartfordSpringfield     39337525    11333012     by units
Houston                152952802    50215064     by units
Indianapolis            18280507    11980810     by units
Jacksonville            19071876     9718127     by units
LasVegas                36323912    18052993     by units
LosAngeles             339632018   168264528     by units
Louisville               9461715     6635288     by units
MiamiFtLauderdale       74156782    23516442     by units
Midsouth               351835659   156513626     by units
Nashville               22644922    12967166     by units
NewOrleansMobile        31426196    14268940     by units
NewYork                172739971    67994156     by units
Northeast              516292515   196988393     by units
NorthernNewEngland      57627868    13905027     by units
Orlando                 41799064    16861640     by units
Philadelphia            47963566    23875232     by units
PhoenixTucson          156735075    38908236     by units
Pittsburgh              11560883     7245462     by units
Plains                 233341172    77847204     by units
Portland                61047851    49504359     by units
RaleighGreensboro       32526051    15676677     by units
RichmondNorfolk         29722621    12508228     by units
Roanoke                 15451017     9590987     by units
Sacramento              66181479     8982268     by units
SanDiego                66449250    23342668     by units
SanFrancisco           120939266    14890924     by units
Seattle                 58522780    50691398     by units
SouthCarolina           42600600    18153173     by units
SouthCentral           777887186   233392434     by units
Southeast              433710759   181527648     by units
Spokane                  9310899     6254375     by units
StLouis                 17336801    14736030     by units
Syracuse                 6750240     4192427     by units
Tampa                   46507848    19496691     by units
TotalUS               4226122969  1638600779     by units
West                   719705525   367073631     by units
WestTexNewMexico       109225323    35296515     by units

In [19]:
df[['Normally buy']].value_counts(normalize=True)

Normally buy
by units        0.981481
by bags         0.018519
dtype: float64

Answer Normally, the customers buy the avocados by unit by 98.1%